projects/observability/src/shared/components/label-detail/label-detail.component.ts

```diff
@@ -0,0 +1,102 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import {
+  ButtonRole,
+  ButtonSize,
+  ButtonStyle,
+  IconSize,
+  PopoverRef,
+  PopoverRelativePositionLocation
+} from '@hypertrace/components';
+
+@Component({
+  selector: 'ht-label-detail',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./label-detail.component.scss'],
+  template: `
+    <ht-event-blocker event="click">
+      <div class="label-detail" *ngIf="this.label">
+        <ht-popover
+          [closeOnClick]="false"
+          [closeOnNavigate]="true"
+          [locationPreferences]="this.locationPreferences"
+          (popoverOpen)="this.onPopoverOpen($event)"
+        >
+          <ht-popover-trigger>
+            <div
+              class="label-detail-trigger"
+              (mouseenter)="this.showdetailButton = true"
+              (mouseleave)="this.showdetailButton = false"
+            >
+              <ht-button
+                *ngIf="this.showdetailButton"
+                class="expand icon detail-button"
+                icon="${IconType.Collapsed}"
+                size="${ButtonSize.Small}"
+                role="${ButtonRole.Tertiary}"
+                display="${ButtonStyle.Text}"
+              ></ht-button>
+              <ht-icon
+                *ngIf="this.icon && !this.showdetailButton"
+                class="icon label-icon"
+                [icon]="this.icon"
+                size="${IconSize.Small}"
+              ></ht-icon>
+              <ht-label [label]="this.label" class="label"></ht-label>
+            </div>
+          </ht-popover-trigger>
+
+          <ht-popover-content>
+            <div class="popover-content">
+              <div class="label-display">
+                <ht-button
+                  class="expanded close"
+                  icon="${IconType.Expanded}"
+                  size="${ButtonSize.Small}"
+                  role="${ButtonRole.Secondary}"
+                  display="${ButtonStyle.Text}"
+                  (click)="this.onClickClose()"
+                ></ht-button>
+                <ht-label [label]="this.label" class="label"></ht-label>
+              </div>
+              <div class="description" *ngIf="this.description">{{ this.description }}</div>
+              <div class="details" *ngIf="this.additionalDetails">
+                <ng-container *ngFor="let detail of this.additionalDetails">
+                  <span class="symbol"></span>
+                  <span class="text">{{ detail }}</span>
+                </ng-container>
+              </div>
+            </div>
+          </ht-popover-content>
+        </ht-popover>
+      </div>
+    </ht-event-blocker>
+  `
+})
+export class LabelDetailComponent {
+  @Input()
+  public icon?: IconType;
+
+  @Input()
+  public label?: string;
+
+  @Input()
+  public description?: string;
+
+  @Input()
+  public additionalDetails?: string[] = [];
```

Comment:
Do we have a screenshot of what this is supposed to be? This seems a strange API. If anything, I'd expect it to take content or a template to show details, rather than a string array.

Reply:
Yes. Screenshot is in description

Reply:
Right now the use case is just a bunch of strings. But I agree, eventually it could be more than this.

Reply:
Got it - the bulleted list still seems a strange API to show details though. Could we change that to content or a template input?

Reply:
What would be the dashboard api look like then? I am still figuring that out. There are two known use case for this. I can explain on slack further

Reply:
Sure - depends on the use case, but any renderer can convert a list of data into content or a template anyway.

Reply:
LGTM. I would prefer to keep this component simple enough unless more use case come up. We try to make components very generic and then, from ux side, we switch to a completely new view. For simple components like these, it is better to iterate when supporting use case come up.

If you want I can move the list div into widget renderer in a **follow up**. Although that doesn't change much from the current state. May be the generic widget should support taking child template render the model inside. But dont want to do that now.

Reply:
I think the problem arises when we make a generic component but don't give it a generic api - if this were a component built for a use case, then it makes sense to name it as such. If it's in our component lib as a label with details, then supporting a generic content for its detail content makes sense - we'll just break the existing usages when we have to update it. We can merge it as is, but this is a principle we need to get better at for exactly the reason you state, ux changes, so we want the api to reflect the concept (an api and a detail view) more than the implementation (a list of details).

Reply:
Although I am not sure how much we want to show in detail for a label. Is a list not enough? Do we think we may have to show a widget inside? It is just supposed to be an info box, which generally only has a text inside.

Reply:
May be i chose a wrong name for the component.

Reply:
how about `label-info` component? It is supposed to show a label with icon and show a text info box popover on click.